### PR TITLE
Use unittest instead of Nose as base

### PR DESCRIPTION
**Pull request by koterpillar**
See #108 for motivation.

### CODE REVIEW

**Comment by koterpillar** [comment]
This finally passes on Python 3, but is still likely to be hard to make work properly on Python 2.

**Comment by koterpillar** [comment]
Passes on Python 2 too. `--no-ignore-python` is broken, and Django and Webdriver need an update too.

File: aloe/loader.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
+"""
+Loader for tests written in Gherkin.
+"""
+
+from __future__ import unicode_literals
```
Comment (danni) [inline aloe/loader.py]: Single line?

File: aloe/loader.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        self.steps_loaded = False
+
+    def load_steps(self):
+        """Discover feature directories and load the step definitions."""
+
+        if self.steps_loaded:
+            return
+
+        self.feature_dirs = [
+            dir_
```
Comment (danni) [inline aloe/loader.py]: Identity comprehension? Could this just be `list()`?

File: aloe/loader.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+            before_all, after_all = CALLBACK_REGISTRY.before_after('all')
+            before_all()
+            self.after_hook = after_all
+
+    def run_after_callbacks(self):
+        """Run the "after all" callbacks."""
+
+        if hasattr(self, 'after_hook'):
+            self.after_hook()
+            delattr(self, 'after_hook')
```
Comment (danni) [inline aloe/loader.py]: This feels weird, why not re-retrieve it from the registry like in `run_before_callbacks`?

File: aloe/main.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
+"""
+Unittest main working with Gherkin tests.
+"""
+
+from __future__ import unicode_literals
```
Comment (danni) [inline aloe/main.py]: Single line?

File: tests/factoryboy_app/features/steps.py [hunk 5]
```diff
@@ -178,4 +177,5 @@ def check_types(self):
                         ('none', type(None)),
                         ('bool', bool),
                         ('date', date)):
-        assert_equal(type(getattr(MyWeirdObject.ref, attr)), type_)
+        # pylint:disable=unidiomatic-typecheck
+        assert type(getattr(MyWeirdObject.ref, attr)) == type_
```
Comment (danni) [inline tests/factoryboy_app/features/steps.py]: `is`

File: aloe/fs.py [hunk 6]
```diff
@@ -37,6 +37,100 @@ def path_to_module_name(filename):
...
+                        mod = old
+                    else:
+                        del sys.modules[part_fqname]
+                        mod = load_module(part_fqname, fh, filename, desc)
+                else:
+                    mod = load_module(part_fqname, fh, filename, desc)
+            finally:
+                if fh:
+                    fh.close()
+                release_lock()
```
Comment (danni) [inline aloe/fs.py:99]: Put some space in here to make the flow control easier to read :)

**Review by danni — COMMENTED** [review]
Generally ++. Minor thoughts.

File: aloe/fs.py (hunk 6 above)
Comment (koterpillar) [reply aloe/fs.py:99]: Lifted from Nose, I'd rather replace this altogether or use some library.

**Review by danni — APPROVED** [review]

**Comment by danni** [comment]
Note to self: I've promised to test this against one of my projects.

**Comment by kingbuzzman** [comment]
what’s the status of this?

**Comment by koterpillar** [comment]
There's no one brave enough to try the change on a live project and share the results here. With the level of invasiveness required to make both nose and, sadly, unittest work, I am reluctant to merge the changes in (see also https://github.com/aloetesting/aloe_django/pull/50) without such a confirmation.

**Comment by kingbuzzman** [comment]
@koterpillar im not going to lie -- yes its a little scary, specially since i just finished creating a migration aloe testing implementation (im going to release it soon) and it heavily relies on nose. I'm going to test test when i get a chance, our code should be the perfect example:

```shell
$ find /code -type f -path '*/features/*' -name '*.py' -exec cat {} \; -exec echo \; | wc -l
   17630
$ find /code -type f -path '*/features/*' -name '*.feature' -exec cat {} \; -exec echo \; | wc -l
   88730
```

**Comment by koterpillar** [comment]
Uncluttering my PRs.

**Pull request closed without merging**